File: binaryTreeLeafsSum.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

node3 = TreeNode(3)
node1 = TreeNode(1)
node6 = TreeNode(6)
node4 = TreeNode(4)

node3.left = node1
node3.right = node6
node6.left = node4

def getTreeLeafsSum(node):
    if node is None:
        return 0
    if node.left is None and node.right is None:
        return node.value
    return getTreeLeafsSum(node.left) + getTreeLeafsSum(node.right)

print getTreeLeafsSum(node3)

File: binaryTreeLeafsSum.js
class TreeNode {
  constructor(value) {
    this.value = value
    this.left = null
    this.right = null
  }
}

const getLeafsSum = node => (node ? (!node.left && !node.right) ? node.value : getLeafsSum(node.left) + getLeafsSum(node.right) : 0)
/*
// normal code here
const getLeafsSum = (node) => {
  if (!node) return 0
  if (!node.left && !node.right) return node.value
  return getLeafsSum(node.left) + getLeafsSum(node.right)
}
*/
const node3 = new TreeNode(3)
const node1 = new TreeNode(1)
const node6 = new TreeNode(6)
const node4 = new TreeNode(4)


node3.left = node1
node3.right = node6
node6.left = node4

console.log(getLeafsSum(node3))